packages/xo-web/src/xo-app/vm/tab-disks.js

```diff
@@ -453,10 +454,20 @@ class AttachDisk extends Component {
         const vbd = vbds[id]
         return !vbd || !vbd.attached || vbd.read_only
       })
-    return attachDiskToVm(vdi, vm, {
-      bootable,
-      mode: readOnly || !_isFreeForWriting(vdi) ? 'RO' : 'RW',
-    }).then(onClose)
+
+    const _attachDisk = () =>
+      attachDiskToVm(vdi, vm, {
+        bootable,
+        mode: readOnly || !_isFreeForWriting(vdi) ? 'RO' : 'RW',
+      }).then(onClose)
+
+    return isEmpty(filter(vbds, { VDI: vdi.id, VM: vm.id, attached: true }))
```

Comment:
You can use `_some`.

If we have an existent VBD which is not connected. Should we duplicate the link between the VDI and the VM?

Reply:
I think, we will inform user when the VBD is already connected because when we `attach disk` we connect the VBD created.

Reply:
We want to show a warning if the selected VDI is already attached to the VM. Are you ok with that?